service.mailnotifiers/notifier.py

```diff
@@ -0,0 +1,271 @@
+# -*- coding: utf-8 -*-
+"""
+kodi service for display number of mail on home screen.
+"""
+
+import xbmc, xbmcgui
+import xbmcaddon
+import poplib, imaplib
+import time
+
+__author__     = "Senufo"
+__scriptid__   = "service.notifier"
+__scriptname__ = "Notifier"
+
+Addon          = xbmcaddon.Addon(__scriptid__)
+
+__cwd__        = Addon.getAddonInfo('path')
+__version__    = Addon.getAddonInfo('version')
+__language__   = Addon.getLocalizedString
+
+__profile__    = xbmc.translatePath(Addon.getAddonInfo('profile'))
+# __resource__   = xbmc.translatePath(os.path.join(__cwd__, 'resources',
+#                                                 'lib'))
+
+DEBUG_LOG = Addon.getSetting('debug')
+if 'true' in DEBUG_LOG: DEBUG_LOG = 1 #loglevel == 1 (DEBUG, shows all)
+else: DEBUG_LOG = -1 #(NONE, nothing at all is logged)
+#DEBUG_LOG = True
+
+# ID HOME window
+WINDOW_HOME = 10000
+
+# Globals Variables
+msg = ''
+NbMsg = [0, 0, 0, 0]
+numEmails = 0
+# No server
+NoServ = 1
+# Text Position
+x = int(Addon.getSetting('x'))
+y = int(Addon.getSetting('y'))
+width = int(Addon.getSetting('width'))
+height = int(Addon.getSetting('height'))
+font = Addon.getSetting('font')
+color = Addon.getSetting('color')
+# Alternate display ??
+ALT = Addon.getSetting('alt')
+# Display in the skin
+SKIN = Addon.getSetting('skin')
+# Display on multilines
+if (Addon.getSetting('multilines').lower == 'true'):
+    SEP = "\n"
+else:
+    SEP = '| '
+# Control ID in window Home.xml
+MsgBox = None
+MsgBoxId = None
+
+start_time = 0
+# Flag for add or not control in HOME
+re_added_control = False
+
+if __name__ == '__main__':
+    monitor = xbmc.Monitor()
+ # Verify if kodi work
+    while not monitor.abortRequested():
+        # Sleep/wait for abort for 10 seconds
+        # Wait before get mails
+        intervalle = int(float(Addon.getSetting('time')) * 60.0)
+        if monitor.waitForAbort(intervalle):
+            # Abort was requested while waiting. We should exit
+            break
+#        xbmc.log("hello addon! %s" % time.time(), level=xbmc.LOGDEBUG)
+
+# Verify if kodi work
+#while (not xbmc.abortRequested):
+    # Wait before get mails
+    #intervalle = int(float(Addon.getSetting('time')) * 60.0)
+        if start_time and (time.time() - start_time) < intervalle:
+            time.sleep(.5)
```

Comment:
Do not use time.sleep, use xbmc.sleep() for values up to 2 secs, beyond that use monitor.waitForAbort().